tests/test_qp.py

```diff
@@ -101,11 +103,21 @@ def test_create_qp_ex_no_attr_connected(self):
             with PD(ctx) as pd:
                 with CQ(ctx, 100, None, None, 0) as cq:
                     qia = get_qp_init_attr_ex(cq, pd, attr, attr_ex, e.IBV_QPT_RC)
-                    with QP(ctx, qia) as qp:
-                        assert qp.qp_state == e.IBV_QPS_RESET, 'RC QP should have been in RESET'
+                    try:
```

Comment:
same comment about extended CQ - It might be more useful for users to get the exact attributes of the QP that failed to create.